src/main/java/com/amazon/opendistroforelasticsearch/sql/metrics/Counter.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.metrics;
+
+public interface Counter<T> {
```

Comment:
Why is this class generic? What types other than `long` do you expect to have in the future? It would be a little odd to have a method `increment` for non-integral types, and all integral types can fit under `long`. Do I miss something?

Reply:
Counter only support integer/long increment, but the value could be float etc., for example, error rate.

Reply:
It seems counter-intuitive to me to have a `float` counter, firstly because I wouldn't know what the increment for `float` is. Seems like it should always be `long`, and then the users can always convert it to whatever numerical type.

Reply:
Not increment for float. I means float for getValue. Anyway, we can make more discussion about it in next release. Thanks.